src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -75,20 +81,23 @@ protected override int PublishMessages(List<QueueWithItems<KafkaItem>> itemsPerT
 
                 } while (true);
             }
-            catch (Exception e)
+            catch (KafkaException e)
             {
                 producer.Dispose();
-
+                if (e.Error.Code == ErrorCode.TransactionalIdAuthorizationFailed)
+                    ThrowTransactionalIdAclAuthorizationException(e);
```

Comment:
This is the only usage of `ThrowTransactionalIdAclAuthorizationException`. Inside this method we're throwing and the beginning of message is the same as here. Can we just adjust the message based on the error code here?

Reply:
done